crates/evm/src/executor/inspector/cheatcodes/fork.rs

```diff
@@ -298,49 +319,75 @@ fn eth_getlogs<DB: DatabaseExt>(data: &EVMData<DB>, inner: &EthGetLogsCall) -> R
         .map_err(|_| fmt_err!("Error in calling eth_getLogs"))?;
 
     if logs.is_empty() {
-        let empty: Bytes = abi::encode(&[Token::Array(vec![])]).into();
+        let empty: Bytes = DynSolValue::Array(vec![]).abi_encode().into();
         return Ok(empty)
     }
 
-    let result = abi::encode(
-        &logs
-            .iter()
+    let result = DynSolValue::Array(
+        logs.iter()
             .map(|entry| {
-                Token::Tuple(vec![
-                    entry.address.into_token(),
-                    entry.topics.clone().into_token(),
-                    Token::Bytes(entry.data.to_vec()),
-                    entry
-                        .block_number
-                        .expect("eth_getLogs response should include block_number field")
-                        .as_u64()
-                        .into_token(),
-                    entry
-                        .transaction_hash
-                        .expect("eth_getLogs response should include transaction_hash field")
-                        .into_token(),
-                    entry
-                        .transaction_index
-                        .expect("eth_getLogs response should include transaction_index field")
-                        .as_u64()
-                        .into_token(),
-                    entry
-                        .block_hash
-                        .expect("eth_getLogs response should include block_hash field")
-                        .into_token(),
-                    entry
-                        .log_index
-                        .expect("eth_getLogs response should include log_index field")
-                        .into_token(),
-                    entry
-                        .removed
-                        .expect("eth_getLogs response should include removed field")
-                        .into_token(),
+                DynSolValue::Tuple(vec![
+                    DynSolValue::Address(entry.address.to_alloy()),
```

Comment:
Should we have `into_token()` impls as well in alloy?

Reply:
https://github.com/foundry-rs/foundry/pull/5928#discussion_r1346568383